sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -274,16 +281,79 @@ protected override async ValueTask<Response> UpdateStatusAsync(bool async, Cance
             }
         }
 
-        private static Response GetResponseFromState(OperationState<T> state)
+        private static Response GetResponseFromState(OperationState<T> state, RequestMethod? requestmethod = null)
         {
             if (state.HasSucceeded)
             {
                 return state.RawResponse;
             }
 
+            // if this is a fake delete lro with 404, just return empty response with 200
+            if (RequestMethod.Delete == requestmethod && state.RawResponse.Status == 404)
+            {
+                return new EmptyResponse(HttpStatusCode.OK);
+            }
+
             throw state.OperationFailedException!;
         }
 
+        private class EmptyResponse : Response
+        {
+            public EmptyResponse(HttpStatusCode status)
+            {
+                Status = (int)status;
+                ReasonPhrase = status.ToString();
+            }
+
+            public override int Status { get; }
+
+            public override string ReasonPhrase { get; }
+
+            public override Stream? ContentStream { get => null; set => throw new System.NotImplementedException(); }
+            public override string ClientRequestId { get => string.Empty; set => throw new System.NotImplementedException(); }
+
+            public override void Dispose()
+            {
+                throw new System.NotImplementedException();
```

Comment:
Do we want to throw here, or just do nothing?

Reply:
same for all other `throw new System.NotImplementedException();` below.

Reply:
yeah, sounds better to have default implementation than throwing, updated.